actors/datacap/src/lib.rs

```diff
@@ -438,17 +453,25 @@ where
     fn resolve_address(&self, addr: &Address) -> Option<ActorID> {
         self.rt.resolve_address(addr)
     }
+
+    fn set_root(&self, _cid: &Cid) -> Result<(), NoStateError> {
+        unimplemented!("self.rt.set_root requires mut but `fvm_actor_utils::Syscalls::set_root` provides non mut")
```

Comment:
The filecoin runtime trait expects non-mut methods for all operations and requires implementations to use interior mutability where state needs to be updated. This usually only necessary in test/mock implementations since the fvm_sdk impl is "stateless" and call out via syscalls to the external VM state. 

I think here, either the builtin-actors runtime should be refactored to follow suit (i.e. provide no `mut self` methods)  or the actor should hold the runtime in a RefCell.

Reply:
Yes @Stebalien and I have discussed this (sorry I can't find where) and agreed the built-in `Runtime` should be refactored to follow suit. Please feel free to file and issue and then make that happen if you're keen.